tools/amp-validation/run.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const amphtmlValidator = require('amphtml-validator');
+const validatorJs = require('./validator-js');
+const fetchPage = require('./fetch-page');
+
+const isDev = process.env.NODE_ENV === 'dev' || false;
+
+module.exports = (endpoints) => {
+    // TODO: re-add pre-release when/if google provide us with one
+    validatorJs.fetchRelease()
+        .then(checkEndpoints(false, endpoints))
+        .catch(onError);
+};
+
+function checkEndpoints(devChannel, endpoints) {
+    return validatorFilePath => amphtmlValidator.getInstance(validatorFilePath)
+        .then(validator => {
+            const tests = endpoints.map(runValidator(validator, devChannel));
+
+            Promise.all(tests)
+                .then(values => {
+                    const exitValue = values.every(Boolean) ? 0 : 1; // every promise returns true <=> exit value is zero
+                    validatorJs.cleanUp();
+                    process.exit(exitValue);
+                });
+        });
+}
+
+function runValidator(validator, devChannel) {
+
+    return endpoint => fetchPage
+        .get(endpoint)
+        .then(res => {
+            console.log(`Checking the AMP validity (${devChannel ? 'pre-release' : 'release'}) of the page at ${isDev ? 'localhost:9000' : ''}${endpoint}, result is:`);
+            const result = validator.validateString(res);
+
+            const pass = result.status === 'PASS';
+            (pass ? console.log : console.error)(`Result for ${endpoint} was: ${result.status}`);
```

Comment:
Changed log/error message here to include the endpoint as it's all async and the messages were appearing out of order sometimes.